### PR TITLE
Sync prepareResult / makePrecise with Java

### PR DESCRIPTION
**Pull request by RZR-UA**

### CODE REVIEW

**Comment by RZR-UA** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2470

File: native/src/routePlannerFrontEnd.cpp [hunk 1]
```diff
@@ -674,11 +684,12 @@ vector<SHARED_PTR<RouteSegmentResult>> RoutePlannerFrontEnd::searchRoute(
 	RoutingContext* ctx, vector<SHARED_PTR<RouteSegmentPoint>>& points,
 	SHARED_PTR<PrecalculatedRouteDirection> routeDirection) {
 	if (points.size() <= 2) {
+		// never reached code
```
Comment (ivanPyrohivskyi) [inline native/src/routePlannerFrontEnd.cpp:687]: redudant

File: native/src/routeSegment.h [hunk 2]
```diff
@@ -201,6 +201,9 @@ struct GpxPoint {
 	GpxPoint(int32_t ind, double lat, double lon, double cumDist) : ind(ind), lat(lat), lon(lon), cumDist(cumDist){};
 
 	GpxPoint(const SHARED_PTR<GpxPoint>& p) : ind(p->ind), lat(p->lat), lon(p->lon), cumDist(p->cumDist){};
+
+	SHARED_PTR<RouteSegmentResult>& getFirstRouteRes() { return routeToTarget.at(0); }
```
Comment (ivanPyrohivskyi) [inline native/src/routeSegment.h:205]: Out of bounds

File: native/src/routeSegment.h [hunk 3]
```diff
@@ -201,6 +201,9 @@ struct GpxPoint {
...
+	SHARED_PTR<RouteSegmentResult>& getLastRouteRes() { return routeToTarget.at(routeToTarget.size() - 1); }
```
Comment (ivanPyrohivskyi) [inline native/src/routeSegment.h:206]: No need &

File: native/src/routePlannerFrontEnd.cpp (hunk 1 above)
Comment (RZR-UA) [reply native/src/routePlannerFrontEnd.cpp:687]: cut

File: native/src/routeSegment.h (hunk 2 above)
Comment (RZR-UA) [reply native/src/routeSegment.h]: done, checked

**Pull request merged**